Assets/MixedRealityToolkit.Tests/PlayModeTests/BoundsControlTests.cs

```diff
@@ -0,0 +1,314 @@
+            bc.center = new Vector3(.25f, 0, 0);
+            bc.size = new Vector3(0.162f, 0.1f, 1);
+            bbox.BoundsOverride = bc;
+            yield return null;
+
+            Bounds b = GetBoundsControlRigBounds(bbox);
+
+            Debug.Assert(b.center == bc.center, $"bounds center should be {bc.center} but they are {b.center}");
+            Debug.Assert(b.size == bc.size, $"bounds size should be {bc.size} but they are {b.size}");
+
+            GameObject.Destroy(bbox.gameObject);
+            GameObject.Destroy(newObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+        }
+
+        /// <summary>
+        /// Uses near interaction to scale the bounds control by directly grabbing corner
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator ScaleViaNearInteration()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            var bounds = bbox.GetComponent<BoxCollider>().bounds;
+            Debug.Assert(bounds.center == new Vector3(0, 0, 1.5f));
+            Debug.Assert(bounds.size == new Vector3(.5f, .5f, .5f));
+
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+
+            // front right corner is corner 3
+            var frontRightCornerPos = bbox.ScaleHandles.Handles[3].transform.position;
+
+            Vector3 initialHandPosition = new Vector3(0, 0, 0.5f);
+            int numSteps = 30;
+            var delta = new Vector3(0.1f, 0.1f, 0f);
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService, ArticulatedHandPose.GestureId.OpenSteadyGrabPoint, initialHandPosition);
+            yield return PlayModeTestUtilities.MoveHandFromTo(initialHandPosition, frontRightCornerPos, numSteps, ArticulatedHandPose.GestureId.OpenSteadyGrabPoint, Handedness.Right, inputSimulationService);
+            yield return PlayModeTestUtilities.MoveHandFromTo(frontRightCornerPos, frontRightCornerPos + delta, numSteps, ArticulatedHandPose.GestureId.Pinch, Handedness.Right, inputSimulationService);
+
+            var endBounds = bbox.GetComponent<BoxCollider>().bounds;
+            TestUtilities.AssertAboutEqual(endBounds.center, new Vector3(0.033f, 0.033f, 1.467f), "endBounds incorrect center");
+            TestUtilities.AssertAboutEqual(endBounds.size, Vector3.one * .567f, "endBounds incorrect size");
+
+            GameObject.Destroy(bbox.gameObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+
+        }
+
+        /// <summary>
+        /// This tests the minimum and maximum scaling for the bounds control.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator ScaleMinMax()
+        {
+            float minScale = 0.5f;
+            float maxScale = 2f;
```

Comment:
It would probably be good to also add a test that changes the bounds and see if the min/max scale adjust accordingly

Reply:
just so i understand this correctly: so your suggestion is testing changing the bounds in code (instead of hand interaction) and checking if the min / max applies to that as well?